packages/notification-user-lifecycle-consumer/src/services/messageProcessor.ts

```diff
@@ -0,0 +1,182 @@
+import { InteropTokenGenerator, logger } from "pagopa-interop-commons";
+import {
+  unsafeBrandId,
+  genericInternalError,
+  generateId,
+} from "pagopa-interop-models";
+import type { UserId, SelfcareId, CorrelationId } from "pagopa-interop-models";
+import { EachMessagePayload } from "kafkajs";
+import { match } from "ts-pattern";
+import { notificationConfigApi } from "pagopa-interop-api-clients";
+import { UsersEventPayload } from "../model/UsersEventPayload.js";
+import { config } from "../config/config.js";
+import { ReadModelServiceSQL } from "./readModelServiceSQL.js";
+import { UserServiceSQL } from "./userServiceSQL.js";
+
+// eslint-disable-next-line max-params
+export async function processUserEvent(
+  payload: UsersEventPayload,
+  readModelServiceSQL: ReadModelServiceSQL,
+  userServiceSQL: UserServiceSQL,
+  notificationConfigProcessClient: ReturnType<
+    typeof notificationConfigApi.createProcessApiClient
+  >,
+  interopTokenGenerator: InteropTokenGenerator,
+  loggerInstance: ReturnType<typeof logger>,
+  correlationId: CorrelationId
+): Promise<void> {
+  const userId = unsafeBrandId<UserId>(payload.user.userId);
+  const institutionId = unsafeBrandId<SelfcareId>(payload.institutionId);
+  const action = payload.eventType;
+  const tenantId = await readModelServiceSQL.getTenantIdBySelfcareId(
+    institutionId
+  );
+
+  if (!tenantId) {
+    throw genericInternalError(
+      `Tenant not found for selfcareId: ${institutionId}`
+    );
+  }
+
+  const userData = {
+    userId,
+    tenantId,
+    institutionId,
+    name: payload.user.name,
+    familyName: payload.user.familyName,
+    email: payload.user.email,
+    productRole: payload.user.productRole,
+  };
+
+  // TODO try catch process call error
+  await match(action)
+    .with("add", async () => {
+      loggerInstance.info(`Add user id ${userId} from tenant ${tenantId}`);
+
+      try {
+        const { serialized } =
+          await interopTokenGenerator.generateInternalToken();
+        await notificationConfigProcessClient.createUserDefaultNotificationConfig(
+          {
+            userId,
+            tenantId,
+          },
+          {
+            headers: {
+              "X-Correlation-Id": correlationId,
+              Authorization: `Bearer ${serialized}`,
+            },
+          }
+        );
+      } catch (err) {
+        loggerInstance.error(
+          `Error creating user default notification config for user ${userId} from tenant ${tenantId}. Reason: ${err}`
+        );
```

Comment:
this logs creation errors but commits the message and does not retry, right? (since it doesn't rethrow)

I should add error handling to #2179 too but I was thinking of logging and discarding the error only if it's a conflict (config already exists) but retrying otherwise